### ISSUE TITLE
File cannot be saved when "the file has modified" dialog is shown

### ISSUE THREAD

**Issue by yanne** [robotframework/RIDE#244]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=209) by bryan.oakley on 13 Aug 2009*

If I open a suite and then change a file on disk, I sometimes get the "The
file has been changed on the file system" dialog twice. I click "No", it
goes away, then it pops right back up. I click "No" again and it goes away.
Then, when I select something different on the tree this happens all over
again. 

So a) it should only pop up once, not twice, and b) once I click "No" it
should somehow record that fact and not ask me again until the file has
been changed on the disk _again_. Once it has changed and I've acknowledged
that fact I shouldn't continue to be told every time I click on a different
item on the tree.

Also, the dialog says "Answering &lt;No&gt; will overwrite the changes on disk"
but that doesn't happen. The changes aren't overwritten until something
causes a save. That warning needs to be worded differently. Even changing
"will" to "may" might be good enough.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=209#c1) by @yanne on 18 Aug 2009*


Preliminary content for 0.18

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=209#c2) by @yanne on 31 Aug 2009*


The reason for this behavior was that the saving silently failed.
Now the overwrite happens as soon as &lt;No&gt; is chosen, and the warning is displayed
only once.

Fix committed in <a href="https://code.google.com/p/robotframework-ride/source/detail?r=695">r695</a>.

It would be cool if you had time to verify this, Bryan.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=209#c4) by @yanne on 7 Sep 2009*


<a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=224" title='The "File modified" query does not work'> Issue 224 </a> has been merged into this issue.

**Comment by yanne** [comment]
> *Originally submitted to [Google Code](http://code.google.com/p/robotframework-ride/issues/detail?id=209#c5) by @yanne on 17 Sep 2009*


<a class="closed_ref" href="https://code.google.com/p/robotframework-ride/issues/detail?id=241" title="File is not saved if it is checked out after the editor is opened"> Issue 241 </a> has been merged into this issue.
